content/posts/cjd-casc.md

```diff
@@ -0,0 +1,548 @@
+
+Once the cluster has been provisioned and `kubectl` has been configured, we'll create a dedicated namespace for our CJD resources and update our `kubectl config` to use it by default:
+
+```bash
+kubectl create namespace cjd
+kubectl config set-context $(kubectl config current-context) --namespace cjd
+```
+
+We'll also need to ensure an ingress controller is deployed within the cluster. For this post, we'll assume the use of the [NGINX ingress controller](https://kubernetes.github.io/ingress-nginx/). Following the [Installation Guide](https://github.com/kubernetes/ingress-nginx/blob/master/docs/deploy/index.md), we'll manually deploy using a few `kubectl` commands:
+
+```bash
+# grant cluster-admin to user
+kubectl create clusterrolebinding cluster-admin-binding \ --clusterrole cluster-admin \ --user $(gcloud config get-value account)
+# deploy nginx ingress controller resources
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/mandatory.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/cloud-generic.yaml
+```
+
+Next, let's look at the manifest file that will deploy the necessary resources for CJD using the [cjd.yaml](https://github.com/cb-technologists/cjd-casc/blob/master/cjd.yaml) manifest file.
+
+First, we create a ServiceAccount, a Role with the necessary permissions to manage agents and perform the required update actions, and a RoleBinding to connect the two.
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cjd
+
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: cjd
+rules:
+- apiGroups: [""]
+  resources: ["pods","configmaps","services","serviceaccounts"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+- apiGroups: [""]
+  resources: ["pods/exec"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+- apiGroups: [""]
+  resources: ["pods/log"]
+  verbs: ["get","list","watch"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get"]
+- apiGroups: ["apps"]
+  resources: ["statefulsets"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+- apiGroups: ["rbac.authorization.k8s.io"]
+  resources: ["roles","rolebindings"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+- apiGroups: ["extensions"]
+  resources: ["ingresses"]
+  verbs: ["create","delete","get","list","patch","update","watch"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: RoleBinding
+metadata:
+  name: cjd
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: cjd
+  # namespace: cjd
+subjects:
+- kind: ServiceAccount
+  name: cjd
+```
+
+Next, we create a Service that exposes ports for access to the web interface and for master-agent communication:
```

Comment:
"Next, we create a Service that exposes ports for access to the **CJD** web interface and for master-agent communication:"